test/aircraft/FlightManagementSystem/RouteModel.spec.js

```diff
@@ -1154,7 +1194,19 @@ ava('._getPastAndPresentLegModels() returns #_previousLegCollection concatenated
     t.deepEqual(result, expectedResult);
 });
 
-ava.todo('._overwriteRouteBetweenWaypointNames()');
+ava('._overwriteRouteBetweenWaypointNames() adjusts divergent/convergent legs and replaces middle content correctly', (t) => {
+    const primaryModel = new RouteModel('KLAS07R.TRALR6.BCE.J11.DRK');
+    const otherModel = new RouteModel('BCE..NAVHO');
+    const expectedResult = [true, { log: 'rerouting to: KLAS07R TRALR6 BCE NAVHO J11 DRK', say: 'rerouting as requested' }];
+    const result = primaryModel._overwriteRouteBetweenWaypointNames('BCE', 'NAVHO', otherModel);
+
+    t.deepEqual(result, expectedResult);
```

Comment:
this assertion belongs in it's own test, break this up into two separate test cases; one for the response and one for the updated `_legCollection`

Reply:
Is it genuinely important to separate this? I definitely don't see this as a problem personally, and would find it redundant to repeat the test verbatim checking something different each time just to get more specificity from the failure of the test (I assume that's the reason to divide it up), and not worth the verbosity, especially since the test's failure message would point you to the issue. I agree in concept that we should want the tests to be pointed and specific, but I would implement that as being generally one test per logic branch, examining all resulting changes/return values/etc for each logical branch. That seems like what we would want, and that's what this test does here by examining the workings of only this private method.